### DISCUSSION TITLE
[Bug Report] setDataSource seems not working when using the field from useField hooks (vue)

### DISCUSSION THREAD

**Discussion by promer94** [alibaba/formily#2115] [Q&A 2.x] [answered]
- [x] I have searched the [issues](https://github.com/alibaba/formily/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
https://codesandbox.io/s/formily-vue-setdatasource-not-work-7p5ig?file=/src/components/MySelect.vue

### Steps to reproduce
check the select options

### What is expected?
select options should change after 3s 

### What is actually happening?
nothing change

### Package
@formily/reactive-vue@2.0.0-rc.4
@formily/vue@2.0.0-rc.4
@formily/core@2.0.0-rc.4
---

### Addtional info
https://codesandbox.io/s/formily-react-setdatasource-work-ck755?file=/App.tsx 

React version works fine

**Comment by muuyao** [comment, marked as answer]
setup只会执行一次，而field的响应式机制datasource变化不会触发options变化，一种方案是直接return field，template 里绑定 field.dataSource
